### PR TITLE
add -p option to type builtin

### PR DESCRIPTION
**Pull request by jyn514**
addresses https://github.com/oilshell/oil/issues/259

if found:
  if file, print the filename
  otherwise do nothing

otherwise, set status to 1

add unit tests for type -p builtin

### CODE REVIEW

File: spec/builtin-bash.test.sh [hunk 1]
```diff
@@ -37,3 +37,16 @@ echo "help=$?"
 cat $TMP/err.txt | grep -i 'no help topics' >/dev/null
 echo "grep=$?"
 ## stdout-json: "help=1\ngrep=0\n"
+
+#### type -p builtin -> file
+type -p mv tar grep
+## stdout-json: "/bin/mv\n/bin/tar\n/bin/grep\n"
```
Comment (andychu) [inline spec/builtin-bash.test.sh]: Please use the newer format here:

```
## STDOUT:
/bin/mv
...
## END
```

For awhile, we didn't have anything but the JSON format, but multiline is more readable for many cases.  I tend to change those when touching older files.

**Comment by andychu** [comment]
Oh sorry I didn't realize this was still open.  It looks ready to merge?

I think I only get a notification when you respond with a comment, so the commits where you addressed the last comment are silent.  Sorry about that!

**Comment by jyn514** [comment]
I think the only thing I haven't address is you wanted the initial commit message changed, I'd have to figure out how `git rebase` works. Other than that I think it's ready to merge, not sure how well it can be extended to `-a` and I haven't tested how the options interact though.

**Comment by andychu** [comment]
OK great I merged it and ran the tests on my machine.  Looks good, thanks!

I forgot to add #259 on the edited change description; oh well.

**Pull request merged**